libkbfs/folder_branch_ops.go

```diff
@@ -797,9 +798,14 @@ func (fbo *folderBranchOps) getMDLocked(
 		// There are no unmerged MDs for this device, so just use the current head.
 		md = mergedMD
 	} else {
-		// We don't need to do this for merged head because the setHeadLocked()
-		// already does that anyway.
-		fbo.setLatestMergedRevisionLocked(ctx, lState, mergedMD.Revision, false)
+		func() {
+			fbo.headLock.Lock(lState)
+			defer fbo.headLock.Unlock(lState)
```

Comment:
Ah, because `TestUnmergedAfterRestart` sets the head via `CheckForNewMDAndInit`.  D'oh.

@songgao: I think `CheckForNewMDAndInit` has to call `GetForTLF` when `md.MergeStatus() == Unmerged`, and set the latest revision with it.